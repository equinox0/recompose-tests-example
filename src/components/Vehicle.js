import React from "react";
import PropTypes from "prop-types";

const Vehicle = ({ vehicle }) => (
    <dl>
        <dt>Model:</dt>
        <dd>{vehicle.model}</dd>
        <dt>Name:</dt>
        <dd>{vehicle.name}</dd>
    </dl>
);

Vehicle.propTypes = {
    vehicle: PropTypes.shape({
        model: PropTypes.string,
        name: PropTypes.string
    }).isRequired
};

export { Vehicle };
export default Vehicle;
