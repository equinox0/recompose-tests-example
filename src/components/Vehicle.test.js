import { shallow } from "enzyme";
import React from "react";
import { Vehicle } from "./Vehicle";

describe("<Vehicle/>", () => {
    describe("component", () => {
        it("should render vehicle", () => {
            const vehicle = {
                model: "Vehicle model",
                name: "Vehicle name"
            };
            const wrapper = shallow(<Vehicle vehicle={vehicle} />);
            expect(wrapper).toMatchSnapshot();
        });
    });
});
