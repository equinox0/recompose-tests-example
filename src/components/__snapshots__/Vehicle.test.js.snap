// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Vehicle/> component should render vehicle 1`] = `
<dl>
  <dt>
    Model:
  </dt>
  <dd>
    Vehicle model
  </dd>
  <dt>
    Name:
  </dt>
  <dd>
    Vehicle name
  </dd>
</dl>
`;
